sysl2/sysl/cmd_export.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	sysl "github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/exporter"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type exportCmd struct {
+	appName string
+	out     string
+	mode    string
+	format  string
+}
+
+const (
+	swaggerMode = "swagger"
+	jsonMode    = "json"
+	yamlMode    = "yaml"
+)
+
+func (p *exportCmd) Name() string            { return "export" }
+func (p *exportCmd) RequireSyslModule() bool { return true }
+
+func (p *exportCmd) Configure(app *kingpin.Application) *kingpin.CmdClause {
+	cmd := app.Command(p.Name(), "Export sysl to external types. Supported types: Swagger")
+	cmd.Flag("app-name", "name of the sysl app defined in sysl model."+
+		" if there are multiple apps defined in sysl model,"+
+		" swagger will be generated only for the given app").Short('a').StringVar(&p.appName)
+	cmd.Flag("format", "format of export, supported options; swagger").Default("swagger").Short('f').StringVar(&p.mode)
+	cmd.Flag("output", "output filepath.format(yaml | json) (default: %(appname).yaml)").Default(
+		"%(appname).yaml").Short('o').StringVar(&p.out)
+	EnsureFlagsNonEmpty(cmd)
+	return cmd
+}
+
+func (p *exportCmd) writeSwaggerForApp(filename string, syslApp *sysl.Application, logger *logrus.Logger) error {
+	var output []byte
+	if p.mode == swaggerMode {
+		swaggerExporter := exporter.MakeSwaggerExporter(syslApp, logger)
+		err := swaggerExporter.GenerateSwagger()
+		if err != nil {
+			logger.Warnf("Error generating swagger for the application %s", err)
+			return err
+		}
+		output, err = swaggerExporter.SerializeOutput(p.format)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("unsupported export format")
+	}
+	dir, err := filepath.Abs(filepath.Dir(filename))
+	if err != nil {
+		return err
+	}
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		logger.Debugf("Target folder exists %s", err)
+		err = os.Mkdir(dir, 0755)
```

Comment:
prefer one line version ```
if err := os.MkDir(); err != nil {
    return err
}```

Here and below (and generally)

Reply:
Nope that doesn't allow you to log that there is actually a permission, it will simply create a directory if that doesn't exist. Besides the Mkdir will fail if the target folder exists.

Reply:
```
if err := os.MkDir(); err != nil { 
   logger.Errorf("Error creating target folder; check permission")
    return err
}```

I dont understand your objection

Reply:
Got that